runtime/interpreter/value.go

```diff
@@ -17868,18 +17874,25 @@ func (v *DictionaryValue) Accept(interpreter *Interpreter, visitor Visitor) {
 }
 
 func (v *DictionaryValue) Iterate(interpreter *Interpreter, f func(key, value Value) (resume bool)) {
+
+	valueComparator := newValueComparator(interpreter, EmptyLocationRange)
+	hashInputProvider := newHashInputProvider(interpreter, EmptyLocationRange)
```

Comment:
`DictionaryValue.Iterate` should ideally be passed the proper location range

Reply:
Hmm, would we need to modify API to add `LocationRange`?

It is currently:
```Go
func (v *DictionaryValue) Iterate(interpreter *Interpreter, f func(key, value Value) (resume bool))
```

Reply:
yes, `DictionaryValue.Iterate` signature should get a new parameter `locationRange LocationRange`, similar to many of the other functions of `Value`s. It might still not be available at each use-site, but we should at least try to provide the right location where available.

Reply:
Done in 5e51834ec43d9a2033115015ca3bc151595e05bf